src/plugins/intel_npu/src/plugin/npuw/lazy_tensor.cpp

```diff
@@ -4,162 +4,190 @@
+#include <variant>
+
+#include "logging.hpp"
+#include "openvino/runtime/make_tensor.hpp"
+#include "util.hpp"
+
 using ov::npuw::weights::LazyTensor;
-using ov::npuw::weights::OrigData;
-using ov::npuw::weights::Transform;
-using ov::npuw::weights::TransformType;
-using ov::npuw::weights::UnpackMeta;
 
 namespace ov {
 namespace npuw {
 namespace weights {
+namespace op {
+struct Const {
+    std::shared_ptr<ov::op::v0::Constant> node;
+
+    std::size_t hash() const {
+        std::size_t seed = std::hash<const void*>()(node->get_data_ptr()) + 0x9e3779b9;
+        seed ^= node->get_element_type().hash() + 0x9e3779b9;
+        for (const auto& dim : node->get_shape()) {
+            seed ^= std::hash<std::size_t>()(dim) + 0x9e3779b9;
+        }
+        return seed;
+    }
+    bool operator==(const Const& other) const {
+        return (node->get_shape() == other.node->get_shape() &&
+                node->get_element_type() == other.node->get_element_type() &&
+                node->get_data_ptr() == other.node->get_data_ptr());
+    }
+    ov::Tensor eval() const {
+        return ov::npuw::util::tensor_from_const(node);
+    }
+};
+struct Concat {
+    std::vector<LazyTensor> tensors;
+    std::size_t axis;
+
+    std::size_t hash() const {
+        std::size_t seed = std::hash<std::size_t>()(axis) + 0x9e3779b9;
+        for (auto& lt : tensors) {
+            seed ^= lt.get_hash() + 0x9e3779b9;
+        }
+        return seed;
+    }
+    bool operator==(const Concat& other) const {
+        return (axis == other.axis && tensors == other.tensors);
+    }
+    ov::Tensor eval() const {
+        std::vector<ov::Tensor> to_concat;
+        for (const auto& lt : tensors) {
+            to_concat.push_back(lt.eval());
+        }
+        return ov::npuw::util::concat(to_concat, axis);
+    }
+};
+
+struct Unpack {
+    LazyTensor w, z, s;
+    ov::element::Type type;
+    ov::Shape shape;
+
+    std::size_t hash() const {
+        std::size_t seed = w.get_hash() + 0x9e3779b9;
+        seed ^= z.get_hash() + 0x9e3779b9;
+        seed ^= s.get_hash() + 0x9e3779b9;
+        seed ^= type.hash() + 0x9e3779b9;
+        for (const auto& dim : shape) {
+            seed ^= std::hash<std::size_t>()(dim) + 0x9e3779b9;
+        }
+        return seed;
+    }
+    bool operator==(const Unpack& other) const {
+        return (type == other.type && shape == other.shape && w == other.w && z == other.z && s == other.s);
+    }
+    ov::Tensor eval() const {
+        const auto& gti = ov::get_tensor_impl;
+        const auto& tw = w.eval();
+        const auto& tz = z.eval();
+        const auto& ts = s.eval();
+        NPUW_ASSERT(tw);
+        ov::Tensor dst(type, shape);
+        if (tw && tz && ts) {
+            ov::npuw::util::unpack(gti(tw), gti(tz), gti(ts), gti(dst));
+        } else if (tw && ts) {
+            ov::npuw::util::unpack(gti(tw), gti(ts), gti(dst));
+        } else {
+            NPUW_ASSERT(false && "Unsupported combination");
```

Comment:
That's the DCOFF without SCALE, why is this unsupported? It is a value cast (e.g, i4 -> f16, i8 -> f16, and so on)

Reply:
so this is not the DCOFF case here - shouldn't be handled at this level yet. Thanks